### ISSUE TITLE
U-he VSTi plugins do not work

### ISSUE THREAD

**Issue by lunadog** [muse-sequencer/muse#365]
The U-he linux native VSTi plugins fail to work as follows:

Crashes plus following error messages:

Hive: Assert failed at /var/lib/jenkins/jobs/u-he/workspace/sources/ObjectLibrary/AM_MipMap.h line 116: "phaseIncrement <= 1.f"

Diva: Assert failed at /var/lib/jenkins/jobs/u he/workspace/sources/Projects/Diva/FilterTunings/DivaFilterTunings.h line 145: "CALLED getTabulatedCoefficients with NULL table!"

ACE: Doesn't crash, but no sound.

Bazille: same as ACE

Podolski - locks up on pressing a note (no sound), but no error message.

Have reported these to u-he list http://www.kvraudio.com/forum/viewtopic.php?f=31&t=424953&sid=ba60bdaac03fd2f048e5ce8239cd674d&start=345 but advised to also report here.

**Comment by terminator356** [comment]
On April 24, 2015 03:00:09 PM lunadog wrote:
> The U-he linux native VSTi plugins fail to work as follows:
> 
> Crashes plus following error messages:
> 
> Hive: Assert failed at
> /var/lib/jenkins/jobs/u-he/workspace/sources/ObjectLibrary/AM_MipMap.h line
> 116: "phaseIncrement <= 1.f"
> 
> Diva: Assert failed at /var/lib/jenkins/jobs/u
> he/workspace/sources/Projects/Diva/FilterTunings/DivaFilterTunings.h line
> 145: "CALLED getTabulatedCoefficients with NULL table!"
> 
> ACE: Doesn't crash, but no sound.
> 
> Bazille: same as ACE
> 
> Podolski - locks up on pressing a note (no sound), but no error message.
> 
> Have reported these to u-he list
> http://www.kvraudio.com/forum/viewtopic.php?f=31&t=424953&sid=ba60bdaac03fd
> 2f048e5ce8239cd674d&start=345 but advised to also report here.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/muse-sequencer/muse/issues/365

Thank you.
I did the native VST support. Will try to investigate, quite a lot on 
 my own plate ATM. Kick me (us) again if nothing happens in a while...

Tim.
The MusE team.

**Comment by lunadog** [comment]
Thanks! Here are some other comments from the same thread:

1. MusE does not call effSetSampleRate and effSetBlockSize before
effMainsChanged in the plugin activation. 
This is the reason for the crashes you mentioned.

2. MusE does not call XInitThreads at the start of main.
This leads to a crash as soon as you try to open Hive's native GUI.

**Comment by spamatica** [comment]
Just a heads up that I'm working on this with very positive results. With the above additions it looks like most of them will work.

**Comment by spamatica** [comment]
Just pushed the the first change that were recommended to master. The second one I believe is not necessary and not recommended.
Back in the day when many VSTi in linux were based on an old version of JUCE this did help a bunch of synths. 
The synths I tried now (mostly Hive, Diva, Podolski but I think I tried all of them) seem to work fine in MusE now, with one caveat. MusE should be built with steinberg vstsdk. All of these synths need chunk support to store their settings correctly and it's only done with the real vstsdk.
